src/shared/functions.py

```diff
@@ -0,0 +1,223 @@
+    with open(file_all_revision_tags, encoding="UTF-8") as rev_tags_f:
+        return rev_tags_f.read().strip().rstrip(",").lstrip(",").split(",")
+
+
+def get_all_releases(file_all_releases):
+    """
+    This function permits to get all the releases.
+    """
+    try:
+        with open(file_all_releases, encoding="UTF-8") as all_releases_fd:
+            return json.load(all_releases_fd)
+    except FileNotFoundError:
+        return {}
+
+
+def get_image_trigger(file_image_trigger):
+    """
+    This function permits to parse and validate the image_trigger file.
+    """
+    with open(file_image_trigger, encoding="UTF-8") as trigger:
+        return ImageSchema(**yaml.safe_load(trigger))
+
+
+def get_tag_mapping_from_all_releases(all_releases_dict):
+    """
+    This function permits to map the tag from all_releases
+    """
+    mapping_from_all_releases = {}
+    for track, risks in all_releases_dict.items():
+        for risk, values in risks.items():
+            if risk in KNOWN_RISKS_ORDERED:
+                tag = f"{track}_{risk}"
+
+                mapping_from_all_releases[tag] = values["target"]
+    return mapping_from_all_releases
+
+
+def get_revision_to_track(all_revision):
+    """
+    This function permits the conversion of the revision to track.
+    """
+    revision_track = {}
+    for track_revision in all_revision:
+        track, revision = track_revision.split("_")
+        if revision in revision_track:
+            msg = (
+                "Each revision can only have 1 canonical tag, "
+                f"but revision {revision} is associated with tracks "
+                f"{track} and {revision_track['revision']}!"
+            )
+            raise BadChannel(msg)
+
+        revision_track[int(revision)] = track
+    return revision_track
+
+
+def get_tag_mapping_from_trigger(image, all_releases_info):
+    """
+    This function permits to get the tag_mapping from image
+    """
+    mapping_from_trigger = {}
+    for track, risks in image.release.items():
+        if track not in all_releases_info:
+            print(f"Track {track} will be created for the 1st time")
+            all_releases_info[track] = {}
+
+        for risk, value in risks.dict(exclude_none=True).items():
+            if risk == "end-of-life":
+                all_releases_info[track]["end-of-life"] = value
+                continue
+
+            if risk not in KNOWN_RISKS_ORDERED:
+                print(f"Skipping unkown risk {risk} in track {track}")
+                continue
+
+            all_releases_info[track][risk] = {"target": value}
+            tag = f"{track}_{risk}"
+            print(f"Channel {tag} points to {value}")
+            mapping_from_trigger[tag] = value
+    return mapping_from_trigger, all_releases_info
+
+
+def get_tag_to_revision(mapping_from_trigger, all_tags_map, revision_track):
+    """
+    This function permits to get the tag to revision.
+    """
+    tag_revision = mapping_from_trigger.copy()
```

Comment:
`tag_revision` you're defining a new var name that you're going to return into `tag_to_revision`. This is confusing and once again, the var name doesn't help understand what its contents are